server/src/main/scala/org/http4s/server/Fallthrough.scala

```diff
@@ -27,7 +27,8 @@ object Fallthrough {
   /** A [[Response]] specific [[Fallthrough]] which considers any response with a 404
     * status code as a fallthrough. */
   implicit def forResponse: Fallthrough[Response] = new Fallthrough[Response] {
-    def isFallthrough(r: Response): Boolean = r.status.code == 404
+    def isFallthrough(r: Response): Boolean =
+      r.status.code == 404 && r.attributes.contains(HttpService.notFoundServiceKey)
```

Comment:
Hmmm, I see your point. Agreed then!

I do prefer renaming the attribute key to something that is specific to fallthrough though.